public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/SpanDetailLinkButtons.tsx

```diff
@@ -0,0 +1,213 @@
+  };
+};
+
+export const getSpanDetailLinkButtons = (props: Props) => {
+  const { span, createSpanLink, traceToProfilesOptions, timeRange, datasourceType, styles } = props;
+
+  let logLinkButton: JSX.Element | null = null;
+  let profileLinkButton: JSX.Element | null = null;
+  let sessionLinkButton: JSX.Element | null = null;
+  if (createSpanLink) {
+    const links = createSpanLink(span);
+    const logsLink = links?.filter((link) => link.type === SpanLinkType.Logs);
+    if (links && logsLink && logsLink.length > 0) {
+      logLinkButton = createLinkButton(logsLink[0], SpanLinkType.Logs, 'Logs for this span', 'gf-logs', datasourceType);
+    }
+    const profilesLink = links?.filter(
+      (link) => link.type === SpanLinkType.Profiles && link.title === RelatedProfilesTitle
+    );
+    if (links && profilesLink && profilesLink.length > 0) {
+      profileLinkButton = createLinkButton(
+        profilesLink[0],
+        SpanLinkType.Profiles,
+        'Profiles for this span',
+        'link',
+        datasourceType,
+        styles?.profilesForThisSpanButton ?? ''
+      );
+    }
+    const sessionLink = links?.filter((link) => link.type === SpanLinkType.Session);
+    if (links && sessionLink && sessionLink.length > 0) {
+      sessionLinkButton = createLinkButton(
+        sessionLink[0],
+        SpanLinkType.Session,
+        'Session for this span',
+        'frontend-observability',
+        datasourceType
+      );
+    }
+  }
+
+  let profileLinkButtons = profileLinkButton;
+  if (profileLinkButton) {
+    // ensure we have a profile link
+    const profilesDrilldownPluginId = 'grafana-pyroscope-app';
+    const context = getProfileLinkButtonsContext(span, traceToProfilesOptions, timeRange);
+
+    // if in explore, use the plugin extension point to get the link
+    // note: plugin extension point links are not currently supported in panel plugins
+    if (window.location.pathname.startsWith('/explore')) {
+      const extensionPointId = PluginExtensionPoints.TraceViewDetails;
+      const { links } = usePluginLinks({ extensionPointId, context, limitPerPlugin: 1 });
+      const link = links && links.length > 0 ? links.find((link) => link.pluginId === profilesDrilldownPluginId) : null;
+
+      // if we have a plugin link, add a button to open in Grafana Profiles Drilldown
+      if (link) {
+        profileLinkButtons = createProfileLinkButtons(profileLinkButton, styles, link);
+      }
+    } else {
+      // fallback to building a url to open in Grafana Profiles Drilldown
+      const drilldownProfilesAppExists = config.apps[profilesDrilldownPluginId];
+
+      if (drilldownProfilesAppExists) {
+        const path = getProfileLinkPath(context, profilesDrilldownPluginId);
+        profileLinkButtons = createProfileLinkButtons(profileLinkButton, styles, undefined, path);
+      }
+    }
+  }
+
+  return { profileLinkButtons, logLinkButton, sessionLinkButton };
+};
+
+export const getProfileLinkButtonsContext = (
```

Comment:
It seems to use a different logic than regular data links where user defines list of tags that should be applied and mapped. We should probably use the same logic to yield consistent results:

<img width="725" alt="Screenshot 2025-03-03 at 15 58 43" src="https://github.com/user-attachments/assets/da327ffd-4dcc-4bb8-ab23-46f2a197bd6c" />